code/game/machinery/telecomms/computers/message.dm

```diff
@@ -80,392 +78,219 @@
+				else
+					error_message = "ALERT: Incorrect decryption key!"
+			return
+		// Return to MSG_MON_SCREEN_MAIN (Main Page)
+		if("return_home")
 			screen = MSG_MON_SCREEN_MAIN
+			return
+		// Deleting message log
+		if("delete_message")
+			linkedServer.pda_msgs -= locate(params["ref"]) in linkedServer.pda_msgs
+			succes_message = "Log Deleted!"
+			return
+		// Deleteing request log
+		if("delete_request")
+			linkedServer.rc_msgs -= locate(params["ref"]) in linkedServer.rc_msgs
+			succes_message = "Log Deleted!"
+			return
+		// Connecting to the server if the old one was broken or destroyed
+		if("connect_server")
+			if(!linkedServer)
+				for(var/obj/machinery/telecomms/message_server/S in GLOB.telecomms_list)
+					linkedServer = S
+					break
+			return
+		// Sending fake (admin) message to crew
+		if("send_fake_message")
+			//Get custom sender
+			var/sender = tgui_input_text(usr, "Please sender's name", "Sender")
+			//Get job
+			var/job = tgui_input_text(usr, "Please job", "Job")
+			//Get recipient
+			var/recipient
+			var/list/viewable_tablets = list()
+			for (var/obj/item/modular_computer/tablet as anything in GLOB.TabletMessengers)
+				var/datum/computer_file/program/messenger/message_app = locate() in tablet.stored_files
+				if(!message_app || message_app.invisible)
+					continue
+				if(!tablet.saved_identification)
+					continue
+				viewable_tablets += tablet
+			if(length(viewable_tablets) > 0)
+				recipient = tgui_input_list(usr, "Select a tablet from the list", "Tablet Selection", viewable_tablets)
+			else
+				recipient = null
+			//Get message
+			var/message = tgui_input_text(usr, "Please enter your message", "Message")
+			if(isnull(sender) || sender == "")
+				sender = "UNKNOWN"
+			// Checking parametrs
+			if(isnull(recipient))
+				notice_message = "NOTICE: No recipient selected!"
+				return attack_hand(usr)
+			if(isnull(message) || message == "")
+				notice_message = "NOTICE: No message entered!"
+				return attack_hand(usr)
+
+			var/datum/signal/subspace/messaging/tablet_msg/signal = new(src, list(
+				"name" = "[sender]",
+				"job" = "[job]",
+				"message" = html_decode(message),
+				"ref" = REF(src),
+				"targets" = list(recipient),
+				"rigged" = FALSE,
+				"automated" = FALSE,
+			))
+			// This will log the signal and transmit it to the target
+			linkedServer.receive_information(signal, null)
+			usr.log_message("(Tablet: [name] | [usr.real_name]) sent \"[message]\" to [signal.format_target()]", LOG_PDA)
+			return
+		// Malfunction AI and cyborgs can hack console. This will auth console, but you need to wait password selection
+		if("hack")
+			var/time = 100 * length(linkedServer.decryptkey)
+			addtimer(CALLBACK(src, PROC_REF(UnmagConsole)), time)
+			screen = MSG_MON_SCREEN_HACKED
+			error_message = "%$&(£: Critical %$$@ Error // !RestArting! <lOadiNg backUp iNput ouTput> - ?pLeaSe wAit!"
+			linkedServer.toggled = FALSE
+			auth = TRUE;
+			return
+	return FALSE
```

Comment:
Don't return FALSE, it means to not update the UI, but we do want to update the UI in an error case because it might fix the bug.